proposal.rst

```diff
@@ -260,18 +351,53 @@ Project Timeline
 .. and/or exams in this time, so we certainly aren't expecting full time work
 .. on your project, but you should aim to complete preliminary work such that
 .. you can actually start coding at the start of the coding period.
-.. 
+..
 .. The coding period is broken into three blocks of about 4 weeks each, with
 .. an evaluation after each block.  The evaluations are to help keep you on
 .. track, and consist of brief evaluation forms sent to GSoC by both the
 .. student and the mentor, and a chance to explicitly review how your project
 .. is going with Xapian mentors.
-.. 
+..
 .. If you will have other commitments during the project time (for example,
 .. any university classes or exams, vacations, etc), make sure you include them
 .. in your project timeline.
 
-FILLME
+
+**Community Bonding Period: (May 4 - June 1) Module 1,2 **
+
+  * Get to know the community and interact with its members
+  * Understand Xapian Code Base, coding conventions and all relevant classes
+  * Understand how the code I am going to add will be structured and what kind of tests i will have to write
+  * Understand which file formats were chosen for previous year's project and why and understand how they were integrated
+  * Discuss with community, other than zip, which file formats are important and are needed to be added to omega.
+
+
+**Coding Weeks 1-3(June 1 - June 20) Module 3**
+
+  * Research about different zip file reading libraries- eg libarchive,libzip ( Compare them and discuss with the community which is the most appropriate)
+  * Implement code to add support for the library
+  * Test code. Fix Issues if any.
+  * Write proper documentation and publish changes to the project
+
+
+**Coding Weeks 4-11(June 22 - Aug 14) Module 4**
+
+During this period I will pick modules one by one and add support for the chosen file format. Each format is estimated to require between 1 and 2 weeks within which the following activities will be carried out:
+
+  * Research about different available libraries (Discuss with the community regarding which is the most effective one)
+  * Implement code to add support for the library
+  * Test code. Fix Issues if any.
+  * Write proper documentation and publish changes to the project.
+
+Support for atleast 5 formats will be added during this period. However, I believe I will be able to add more in the given time frame.
+
+A week has been kept free as a buffer in case of any delay in any of the modules.
+
+By the first evaluation (July 29 - July 3), I will hopefully complete module 3
+
+By the second evaulation (July 27 - July 31), I plan to complete adding support for 2 more file formats.
+
+And by Aug 24, I will add support for 2 more, Hopefully I will be able to add support for more file formats as discussed(Module 5)
```

Comment:
I'd really like to see this broken down more by time.  Currently your coding period is split into 2 sections of 3 weeks and 8 weeks, but nothing is mapped out in smaller units of time.  That means that most of the time during the coding period we can't tell how on track we are easily, which is far from ideal.

Also, how have you come up with these lengths of time?  My experience is that people are bad at estimating time for large tasks but can do a reasonable job of estimating small tasks, so breaking up a large task into smaller subtasks until you reach a level of detail you are more familiar with is a good way to get better time estimates.  It also forces you to think more about the details, and the details are usually what actually takes a lot of the time.

Reply:
Thanks for this detailed comment. I tried to use a top-down approach to build a timeline, but due to the uncertainty over the details in the modules, I ended up making two large blocks instead of a more granular level of detail. I’ve updated the timeline to improve this. Please let me know Phase 1 is okay, and if there are other changes I might make to improve its robustness. But I’m having trouble with Phases 2 and 3 (in the updated timeline) because it is unclear which libraries we will add and also because whichever libraries we decide will mostly follow a similar process as Phase 1. Except that it would now take less time for each new library since I would be more familiar with it. So I’m confused about how I should present this. If you could provide some guidance on how I can add more details, that would be very helpful.

Reply:
I think you're along the right lines there - assuming the libraries don't have additional challenges, you're likely to be a bit quicker for each subsequent one (but plateauing eventually).

I don't think you can do better unless you identify some candidate libraries now and try to assess if they are likely to present additional challenges.

Reply:
> I don't think you can do better unless you identify some candidate libraries now and try to assess if they are likely to present additional challenges.


Thank you for pointing this out, I have tried and identified few file formats which can be worked upon. I have now updated phases 2 and 3 in the timeline with rational with why I am proposing these libraries (these are subject to change), please let me know if this makes sense or should I change something.



> As olly says, the main thing is you need to break down the timeline more. One suggestion that may help to do this: for the one module (libarchive) you've identified, break that one down (into sections of _at most_ one week) and basically say "this is a template for adding support for a number of file formats using one library". Then you can simply identify two or three other libraries that you think are useful. (And it's fine if after later discussion you change which ones they'll be.) That will at least give you a clear timeline to work against, and enable you to think in detail about what steps you'll have to go through to add new formats.

Thanks for the detailed comment. I have tried and broken down Phase 1 into sections and have tried to elaborate these individual sections. If you have any suggestions about the details that I have listed in Phase 1, please let me know, I will work on them.